#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.33905928649454575 0.8840566267578355 0.3216872408526127 4.647555435862453
  position -0.4486073195903859 1.448803498489871 0.5938418783391346
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0.96 0 0.54
  floorSize 1.92 1.08
  wallHeight 0.01
}
Robot {
  translation 1.04 0.035 0.44
  rotation -0.9999999992145252 1.0103619340882666e-10 3.963520528086543e-05 -2.4281047995477075e-08
  children [
    Solid {
      children [
        DEF BotBody Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0
            metalness 0
          }
          geometry Box {
            size 0.13 0.05 0.09
          }
        }
      ]
      boundingObject USE BotBody
      physics Physics {
      }
    }
    Track {
      translation 0 0 0.05
      children [
        DEF G0 Group {
          children [
            DEF BotTrackBox Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.095 0.06 0.02
              }
            }
            Transform {
              translation -0.05 0 0
              rotation 1 0 0 -1.5707953071795862
              children [
                DEF BotTrackCyl Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.03
                  }
                }
              ]
            }
            Transform {
              translation 0.05 0 0
              rotation -1 0 0 1.5708
              children [
                USE BotTrackCyl
              ]
            }
          ]
        }
      ]
      name "trackRight"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "RightMotor"
        }
      ]
    }
    Track {
      translation 0 0 -0.05
      children [
        USE G0
      ]
      name "trackLeft"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "LeftMotor"
        }
      ]
    }
    Compass {
      translation -3.81029e-14 -7.79238e-15 -2.59099e-14
      rotation -0.9658216086544755 -0.18391592547829366 -0.18265692598843322 3.67802e-06
    }
    GPS {
      accuracy 0.001
    }
  ]
  name "red"
  boundingObject USE BotBody
  physics Physics {
  }
  controller "RobotController3"
}
Robot {
  translation 1.14 0.035 0.75
  rotation -0.9999999992145252 1.0103619340882666e-10 3.963520528086543e-05 -2.4281047995477075e-08
  children [
    Solid {
      children [
        DEF BotBody Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            metalness 0
          }
          geometry Box {
            size 0.13 0.05 0.09
          }
        }
      ]
      boundingObject USE BotBody
      physics Physics {
      }
    }
    Track {
      translation 0 0 0.05
      children [
        DEF G0 Group {
          children [
            DEF BotTrackBox Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.095 0.06 0.02
              }
            }
            Transform {
              translation -0.05 0 0
              rotation 1 0 0 -1.5707953071795862
              children [
                DEF BotTrackCyl Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.03
                  }
                }
              ]
            }
            Transform {
              translation 0.05 0 0
              rotation -1 0 0 1.5708
              children [
                USE BotTrackCyl
              ]
            }
          ]
        }
      ]
      name "trackRight"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "RightMotor"
        }
      ]
    }
    Track {
      translation 0 0 -0.05
      children [
        USE G0
      ]
      name "trackLeft"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "LeftMotor"
        }
      ]
    }
    Compass {
      translation -3.81029e-14 -7.79238e-15 -2.59099e-14
      rotation -0.9658216086544755 -0.18391592547829366 -0.18265692598843322 3.67802e-06
    }
    GPS {
      accuracy 0.001
    }
  ]
  name "green"
  boundingObject USE BotBody
  physics Physics {
  }
  controller "RobotController4"
}
Robot {
  translation 1.2400000000000002 0.035 0.8999999999999999
  rotation -0.9999999992145252 1.0103619340882666e-10 3.963520528086543e-05 -2.4281047995477075e-08
  children [
    Solid {
      children [
        DEF BotBody Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.13 0.05 0.09
          }
        }
      ]
      boundingObject USE BotBody
      physics Physics {
      }
    }
    Track {
      translation 0 0 0.05
      children [
        DEF G0 Group {
          children [
            DEF BotTrackBox Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.095 0.06 0.02
              }
            }
            Transform {
              translation -0.05 0 0
              rotation 1 0 0 -1.5707953071795862
              children [
                DEF BotTrackCyl Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.03
                  }
                }
              ]
            }
            Transform {
              translation 0.05 0 0
              rotation -1 0 0 1.5708
              children [
                USE BotTrackCyl
              ]
            }
          ]
        }
      ]
      name "trackRight"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "RightMotor"
        }
      ]
    }
    Track {
      translation 0 0 -0.05
      children [
        USE G0
      ]
      name "trackLeft"
      boundingObject USE G0
      physics Physics {
      }
      device [
        LinearMotor {
          name "LeftMotor"
        }
      ]
    }
    Compass {
      translation -3.81029e-14 -7.79238e-15 -2.59099e-14
      rotation -0.9658216086544755 -0.18391592547829366 -0.18265692598843322 3.67802e-06
    }
    GPS {
      accuracy 0.001
    }
  ]
  name "blue"
  boundingObject USE BotBody
  physics Physics {
  }
  controller "RobotController5"
}
